### PR TITLE
Base class for extensions.

### PR DESCRIPTION
**Pull request by jorenham**
I also changed some method signatures to match the base class

### CODE REVIEW

File: scrapy/extensions/closespider.py [hunk 1]
```diff
@@ -32,7 +33,7 @@ def __init__(self, crawler):
         if self.close_on.get('errorcount'):
             crawler.signals.connect(self.error_count, signal=signals.spider_error)
         if self.close_on.get('pagecount'):
-            crawler.signals.connect(self.page_count, signal=signals.response_received)
+            crawler.signals.connect(self.response_received, signal=signals.response_received)
```
Comment (Eauwzeauw) [inline scrapy/extensions/closespider.py:36]: why did you rename this? (also method declaration below). It does the same, but you think this is more clear? Should you not also change 'pagecount' in
`: if self.close_on.get('pagecount'):`
then?

**Review by Eauwzeauw — APPROVED** [review]
Looks good, let's hope also they think so:)

File: scrapy/extensions/closespider.py (hunk 1 above)
Comment (jorenham) [reply scrapy/extensions/closespider.py:36]: This is the callback for the `response_received` signal. In `BaseExtension` this method is aptly named `response_received`. So for consistency I renamed it and now it overrides the correct base class method.

**Review by jojona — APPROVED** [review]

**Review by Eauwzeauw — APPROVED** [review]
